distributed/profile.py

```diff
@@ -314,16 +317,13 @@ def traverse(state, start, stop, height):
     }
 
 
-_watch_running: set[int] = set()
-
-
 def wait_profiler() -> None:
```

Comment:
Should we refactor existing tests to explicitly use the locking mechanism instead?

Reply:
Yes, I think that would be good